File: src/main/java/net/nextpulse/jacumulus/responses/AcumulusResponse.java
package net.nextpulse.jacumulus.responses;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlElementWrapper;
import java.util.List;

@XmlAccessorType(XmlAccessType.FIELD)
public abstract class AcumulusResponse {
  
  @XmlElementWrapper(name = "errors")
  @XmlElement(name = "error")
  private List<ApiMessage> errors;
  
  @XmlElementWrapper(name = "warnings")
  @XmlElement(name = "warning")
  private List<ApiMessage> warnings;
  
  public List<ApiMessage> getErrors() {
    return errors;
  }
  
  public void setErrors(List<ApiMessage> errors) {
    this.errors = errors;
  }
  
  public List<ApiMessage> getWarnings() {
    return warnings;
  }
  
  public void setWarnings(List<ApiMessage> warnings) {
    this.warnings = warnings;
  }
}

File: src/main/java/net/nextpulse/jacumulus/responses/AcumulusResponse.kt
package net.nextpulse.jacumulus.responses

import net.nextpulse.jacumulus.responses.models.ApiMessage
import javax.xml.bind.annotation.XmlElement
import javax.xml.bind.annotation.XmlElementWrapper

/**
 * Parent class for all API responses
 */
abstract class AcumulusResponse {

    @get:XmlElementWrapper(name = "errors")
    @get:XmlElement(name = "error")
    var errors: List<ApiMessage> = ArrayList()

    @get:XmlElementWrapper(name = "warnings")
    @get:XmlElement(name = "warning")
    var warnings: List<ApiMessage> = ArrayList()
}
